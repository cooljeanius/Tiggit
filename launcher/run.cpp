#include "run.hpp"

#include <stdexcept>

void noImpl(const std::string &command, const std::string &workdir = "")
{
  throw std::runtime_error("Cannot run " + command + ": Launching is not yet implemented on this platform");
}

#ifdef _WIN32
#include "run_windows.hpp"
#define RUN win32_run
#else
#define RUN noImpl
#endif

void Launcher::run(const std::string &command, const std::string &workdir)
{
  RUN (command, workdir);
}
