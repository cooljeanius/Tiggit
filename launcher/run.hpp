#ifndef __LAUNCHER_RUN_HPP_
#define __LAUNCHER_RUN_HPP_

#include <string>

namespace Launcher
{
  /* Run the given command in the specified working directory. If no
     dir is specified, use the current process' working dir.
   */
  void run(const std::string &command, const std::string &workdir = "");
}

#endif
