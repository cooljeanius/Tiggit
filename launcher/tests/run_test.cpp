#include "run.hpp"

#include <iostream>

using namespace std;
using namespace Launcher;

int main(int argc, char **argv)
{
  if(argc != 1)
    {
      cout << "Got arguments, exiting.\n";
      return 1;
    }

  cout << "Running ourselves\n";
  try { run("run_test"); }
  catch(exception &e)
    {
      cout << "CAUGHT: " << e.what() << endl;
    }
  catch(...)
    {
      cout << "An unknown error occured\n";
    }

  return 0;
}
