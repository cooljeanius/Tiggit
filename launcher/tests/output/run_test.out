Running ourselves
CAUGHT: Cannot run run_test: Launching is not yet implemented on this platform
